packages/runtime/container-runtime/src/summaryManager.ts

```diff
@@ -45,12 +52,103 @@ enum SummaryManagerState {
+ * It will handle updated the elected client when a summary ack hasn't been seen
+ * for some configured number of ops.
+ */
+export class ElectedSummarizerManager {
+    private readonly clientElection: IOrderedClientElection;
+    /**
+     * Used to calculate number of ops since last summary ack for the current elected client.
+     * This will be undefined if we do not have enough information or there is no elected summarizer.
+     * When a new client is first elected, this will be initially set to the sequence number of
+     * the first inbound op after they are elected.
+     * When a summary ack comes in, this will be set to the sequence number of the summary ack.
+     */
+    private lastSummaryAckSeqForClient: number | undefined;
+    /** Most recently invalidated elected summarizer client id */
+    private invalidClientId: string | undefined;
+
+    public get electedClientId() {
+        return this.clientElection.electedClient?.clientId;
+    }
+
+    constructor(
+        private readonly logger: ITelemetryLogger,
+        private readonly summaryCollection: IEventProvider<ISummaryCollectionOpEvents>,
+        orderedClientCollection: IOrderedClientCollection,
+        private readonly maxOpsSinceLastSummary: number,
+        private readonly refreshSummarizer: () => void,
+    ) {
+        // On every inbound op, if enough ops pass without seeing a summary ack (per elected client),
+        // elect a new client and log to telemetry.
+        this.summaryCollection.on("default", (op) => {
+            if (this.lastSummaryAckSeqForClient === undefined) {
+                // We don't have enough information to know if a client is not summarizing yet.
+                if (this.electedClientId !== undefined) {
+                    this.lastSummaryAckSeqForClient = op.sequenceNumber;
+                }
+                return;
+            }
+            const opsSinceLastAckForClient = op.sequenceNumber - this.lastSummaryAckSeqForClient;
+            const electedClientId = this.electedClientId;
+            if (
+                opsSinceLastAckForClient > this.maxOpsSinceLastSummary
+                && electedClientId !== undefined
+                && electedClientId !== this.invalidClientId
+            ) {
+                this.logger.sendErrorEvent({
+                    eventName: "ElectedClientNotSummarizing",
+                    electedClientId,
+                    sequenceNumber: op.sequenceNumber,
+                    lastSummaryAckSeqForClient: this.lastSummaryAckSeqForClient,
+                });
+                this.invalidClientId = electedClientId;
+
+                // Elect a new summarizer client.
+                this.clientElection.incrementElectedClient();
+            }
+        });
+
+        // When a summary ack comes in, reset our op seq counter.
+        this.summaryCollection.on(MessageType.SummaryAck, (op) => {
+            this.invalidClientId = undefined;
```

Comment:
should we also reset our electedClientId if needed as given the above we might not even know who he summarizer should be right now